diff --git a/dist/instagram.js b/dist/instagram.js
index 6bb6eaa84127ebdebd1933d4bc5bcf50fa6bdda5..c019696e8209e9ff0c1726421e729d18f071b03f 100644
--- a/dist/instagram.js
+++ b/dist/instagram.js
@@ -4,15 +4,14 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.instagramGetUrl = instagramGetUrl;
-const axios_1 = __importDefault(require("axios"));
+exports.instagramGetId = getShortcode;
+exports.instagramGetRedirect = checkRedirect;
 const qs_1 = __importDefault(require("qs"));
 //Main function
-async function instagramGetUrl(url_media, config = { retries: 5, delay: 1000 }) {
+async function instagramGetUrl(id_media, config = { retries: 5, delay: 1000 }) {
     return new Promise(async (resolve, reject) => {
         try {
-            url_media = await checkRedirect(url_media);
-            const SHORTCODE = getShortcode(url_media);
-            const INSTAGRAM_REQUEST = await instagramRequest(SHORTCODE, config.retries, config.delay);
+            const INSTAGRAM_REQUEST = await instagramRequest(id_media, config.retries, config.delay, config.proxy);
             const OUTPUT_DATA = createOutputData(INSTAGRAM_REQUEST);
             resolve(OUTPUT_DATA);
         }
@@ -22,11 +21,14 @@ async function instagramGetUrl(url_media, config = { retries: 5, delay: 1000 })
     });
 }
 //Utilities
-async function checkRedirect(url) {
+async function checkRedirect(url, proxy) {
     let split_url = url.split("/");
     if (split_url.includes("share")) {
-        let res = await axios_1.default.get(url);
-        return res.request.path;
+		const response = await fetch(url, {
+			method: 'GET',
+			proxy,
+		})
+        return response.url;
     }
     return url;
 }
@@ -83,19 +85,17 @@ function getShortcode(url) {
         throw new Error(`Failed to obtain shortcode: ${err.message}`);
     }
 }
-async function getCSRFToken() {
+async function getCSRFToken(proxy) {
     try {
-        let config = {
-            method: 'GET',
-            url: 'https://www.instagram.com/',
-        };
         const token = await new Promise((resolve, reject) => {
-            axios_1.default.request(config).then((response) => {
-                if (!response.headers['set-cookie']) {
+            fetch('https://www.instagram.com/', {
+            	method: 'GET',
+             	proxy,
+            }).then((response) => {
+                if (!response.headers.get('set-cookie')) {
                     reject(new Error('CSRF token not found in response headers.'));
-                }
-                else {
-                    const csrfCookie = response.headers['set-cookie'][0];
+                } else {
+                    const csrfCookie = response.headers.get('set-cookie')[0];
                     const csrfToken = csrfCookie.split(";")[0].replace("csrftoken=", '');
                     resolve(csrfToken);
                 }
@@ -117,7 +117,7 @@ function isSidecar(requestData) {
         throw new Error(`Failed sidecar verification: ${err.message}`);
     }
 }
-async function instagramRequest(shortcode, retries, delay) {
+async function instagramRequest(shortcode, retries, delay, proxy) {
     var _a;
     try {
         const BASE_URL = "https://www.instagram.com/graphql/query";
@@ -131,18 +131,17 @@ async function instagramRequest(shortcode, retries, delay) {
             }),
             'doc_id': INSTAGRAM_DOCUMENT_ID
         });
-        const token = await getCSRFToken();
-        let config = {
-            method: 'post',
-            maxBodyLength: Infinity,
-            url: BASE_URL,
-            headers: {
-                'X-CSRFToken': token,
-                'Content-Type': 'application/x-www-form-urlencoded',
-            },
-            data: dataBody
-        };
-        const { data } = await axios_1.default.request(config);
+        const token = await getCSRFToken(proxy);
+        const response = await fetch(BASE_URL, {
+        	proxy,
+        	method: 'POST',
+         	headers: {
+            	'X-CSRFToken': token,
+            	'Content-Type': 'application/x-www-form-urlencoded',
+          	},
+           	body: dataBody
+        })
+        const data = await response.json();
         if (!((_a = data.data) === null || _a === void 0 ? void 0 : _a.xdt_shortcode_media))
             throw new Error("Only posts/reels supported, check if your link is valid.");
         return data.data.xdt_shortcode_media;
diff --git a/dist/types/instagram.d.ts b/dist/types/instagram.d.ts
index a132503ece1623a404c3613f2340eb44a5993cb1..61eb385bb1a646e3b63ff20a20ba370da9dc5e2e 100644
--- a/dist/types/instagram.d.ts
+++ b/dist/types/instagram.d.ts
@@ -24,7 +24,13 @@ export interface InstagramResponse {
 export interface InstagramError {
     error: string;
 }
-export declare function instagramGetUrl(url_media: string, config?: {
+
+export declare function instagramGetUrl(id_media: string, config?: {
     retries: number;
     delay: number;
+	proxy?: string;
 }): Promise<InstagramResponse>;
+
+export declare function instagramGetId(url_media:string): Promise<string>;
+
+export declare function instagramGetRedirect(url_media:string, proxy?: string): Promise<string>;
diff --git a/package.json b/package.json
index bf293eee1fe2d4b956e97614972c773620ceb90a..003c8ca8fdb5b21ede47811255385a28ba0df7f1 100644
--- a/package.json
+++ b/package.json
@@ -43,7 +43,6 @@
   },
   "homepage": "https://github.com/victorsouzaleal/instagram-direct-url#readme",
   "dependencies": {
-    "axios": "^1.7.7",
     "qs": "^6.13.0"
   },
   "devDependencies": {
